cloud_function/main.py

```diff
@@ -0,0 +1,127 @@
+import os
+import json
+import requests
+from google.cloud import storage
+from google.cloud import exceptions
+
+project = os.environ.get("GCP_PROJECT", "broad-gotc-dev")
+wfl_url = os.environ.get("WFL_URL", "https://workflow-launcher.gotc-dev.broadinstitute.org")
+default_credentials = os.environ.get("DEFAULT_CREDENTIALS", False)
+
+
+def get_test_credentials():
+    import google.auth
+    import google.auth.transport.requests
+
+    scopes = ["https://www.googleapis.com/auth/cloud-platform",
+              "https://www.googleapis.com/auth/userinfo.email",
+              "https://www.googleapis.com/auth/userinfo.profile"]
+    credentials, gcp_project = google.auth.default(scopes=scopes)
+    if not credentials.valid:
+        credentials.refresh(google.auth.transport.requests.Request())
+    return credentials
+
+def get_auth_headers():
+    scopes_list = ["https://www.googleapis.com/auth/cloud-platform",
+                   "https://www.googleapis.com/auth/userinfo.email",
+                   "https://www.googleapis.com/auth/userinfo.profile"]
+    scopes = ",".join(scopes_list)
+    metadata_url = f'http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token?scopes={scopes}'
+    metadata_headers = {'Metadata-Flavor': 'Google'}
+    r = requests.get(metadata_url, headers=metadata_headers)
+    r.raise_for_status()
+    access_token = r.json()['access_token']
+    headers = {
+        'Authorization': 'Bearer {}'.format(access_token)
+    }
+    return headers
+
+def get_manifest_path(object_name):
+    parts = object_name.split("/")
+    chipwell_barcode = parts[0]
+    analysis_version = parts[1]
```

Comment:
```suggestion
    chipwell_barcode, analysis_version = object_name.split("/")
```

Reply:
There would be more than two items in the "parts" array so this would throw an error. 
(e.g. if the object name is "chipwell_barcode/analysis_version/some_data.json")

Reply:
Also I realized there is a "/" in front of the file name in the event payload so I need to fix this parsing 🤦‍♀️

Reply:
> There would be more than two items in the "parts" array so this would throw an error.
> (e.g. if the object name is "chipwell_barcode/analysis_version/some_data.json")

Oh I see, my bad